Panic on ARM

Hi,

a user of cluster reports a panic coming from pubsub on a BeagleBone:

https://github.com/ipfs/ipfs-cluster/issues/433

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x4 pc=0x11540]

goroutine 46 [running]:
sync/atomic.addUint64(0x131c005c, 0x1, 0x0, 0x131c9150, 0x8)
	/usr/lib64/go/1.9/src/sync/atomic/64bit_arm.go:31 +0x4c
gx/ipfs/QmduQtUFqdq2RhM84yM2mYsdgJRAH8Aukb15viDxWkZtvP/go-libp2p-floodsub.(*PubSub).Publish(0x131c0000, 0x716b54, 0xf, 0x13124630, 0x5d, 0xa2, 0x0, 0x1e)
	gx/ipfs/QmduQtUFqdq2RhM84yM2mYsdgJRAH8Aukb15viDxWkZtvP/go-libp2p-floodsub/pubsub.go:556 +0x64
github.com/ipfs/ipfs-cluster/monitor/pubsubmon.(*Monitor).PublishMetric(0x12d5caa0, 0x70dc03, 0x4, 0x12d16c60, 0x22, 0x0, 0x0, 0xa5f1d075, 0x1531f22e, 0x1, ...)
	github.com/ipfs/ipfs-cluster/monitor/pubsubmon/pubsubmon.go:193 +0x39c
github.com/ipfs/ipfs-cluster.(*Cluster).pushPingMetrics(0x12d64320)
	github.com/ipfs/ipfs-cluster/cluster.go:232 +0xc0
created by github.com/ipfs/ipfs-cluster.(*Cluster).run
	github.com/ipfs/ipfs-cluster/cluster.go:340 +0x48
```

Might this come from the warning here: https://golang.org/pkg/sync/atomic/#pkg-note-BUG ? @vyzo 

I'll be testing with my own RPis next week, maybe I can provide more info.

--- a/pubsub.go
+++ b/pubsub.go
@@ -26,6 +26,13 @@ const (
 var log = logging.Logger("pubsub")
 
 type PubSub struct {
+	// atomic counter for seqnos
+	// NOTE: Must be declared at the top of the struct as we perform atomic
+	// operations on this field.
+	//
+	// See: https://golang.org/pkg/sync/atomic/#pkg-note-BUG
+	counter uint64
+
 	host host.Host
 
 	rt PubSubRouter
@@ -79,9 +86,6 @@ type PubSub struct {
 	seenMessages *timecache.TimeCache
 
 	ctx context.Context
-
-	// atomic counter for seqnos
-	counter uint64
 }
 
 // PubSubRouter is the message router component of PubSub

